src/app/user-details/user-details.component.html

```diff
@@ -0,0 +1,24 @@
+
+<div>
+  <h4>id: {{ userinfo.id }}</h4>
+  <h4>Name: {{ userinfo.name }}</h4>
+  <h4>Surname: {{ userinfo.surname }}</h4>
+  <h4>Age: {{ userinfo.age }}</h4>
+  <h4>Date of birth: {{ userinfo.dateOfBirth | date}}</h4>
+  <h3>Address:</h3>
+  <div *ngFor="let c in userinfo.address">
```

Comment:
c non ha molto significato, devi dare dei nomi migliori
non puoi ciclare su un oggetto, non in questo caso
c.key e c.value scritti così non vanno bene, in questo caso forse avresti dovuto utilizzare una pipe apposita, ma la cosa migliore era un semplice binding